tests/core/stack/test_stack.py

```diff
@@ -63,59 +71,79 @@ def test_dup_does_not_allow_stack_to_exceed_1024_items(stack):
 
 
 @pytest.mark.parametrize(
-    ("items,type_hint"),
+    ("items, stack_method"),
     (
-        ([1], UINT256),
-        ([1, 2, 3], UINT256),
-        ([b'1', b'10', b'101', b'1010'], BYTES)
+        ([1], 'push_int'),
+        ([1, 2, 3], 'push_int'),
+        ([b'1', b'10', b'101', b'1010'], 'push_bytes')
     )
 )
-def test_pop_returns_latest_stack_item(stack, items, type_hint):
+def test_pop_returns_latest_stack_item(stack, items, stack_method):
+    method = getattr(stack, stack_method)
     for each in items:
-        stack.push(each)
-    assert stack.pop(num_items=1, type_hint=type_hint) == items[-1]
+        method(each)
+    assert stack.pop1_any() == items[-1]
 
 
 @pytest.mark.parametrize(
-    ("value,type_hint,type,is_valid"),
+    ("value, push_method, pop_method, expect_result"),
     (
-        (1, UINT256, int, True),
-        (b'101', BYTES, bytes, True),
-        (1, SECPK1_N, int, False)
+        (1, 'push_int', 'pop_ints', (1, )),
+        (1, 'push_int', 'pop_any', (1, )),
+        (1, 'push_int', 'pop_bytes', (b'\x01', )),
+        (1, 'push_int', 'pop1_int', 1),
+        (1, 'push_int', 'pop1_any', 1),
+        (1, 'push_int', 'pop1_bytes', b'\x01'),
```

Comment:
I don't know how I feel about this, but I realized you could do this as:


```
@pytest.parametrize(
    "method",
    (
        SomeClass.method_a,
        SomeClass.method_b,
    ),
)
de test_some_methods_on_a_class(method):
    instance = SomeClass(...)
    result = method(instance, ...)
    assert result ...
```

Leverage providing the `self` argument and call the method on the class itself.

Reply:
Eh, seems about the same to me. The slight gain in avoiding the `getattr()` is offset by the longer parameterization boilerplate. Not opposed, but probably not worth doing the change now.

Reply:
Wasn't really advocating for it as much as realizing it was possible.  :+1: